dupin/detect/dynp.py

```diff
@@ -0,0 +1,52 @@
+"""Implements dynamic programming class for optimal segementation algorithm."""
+import _DynP
+import numpy as np
+
+
+class DynP:
+    """Detects the change points in a time series.
+
+    Attributes
+    ----------
+    data: np.ndarray
+        Matrix storing the time series data.
+    num_bkps: int
+        Number of change points to detect.
+    jump: int
+        Interval for checking potential change points. Changing will
+        not provide optimal detection, but will reduce runtime.
+    min_size: int
+        Minimum size of a segment. Changing will not provide optimal
+        detection, but will reduce runtime.
+    """
+
+    def __init__(
+        self, data: np.ndarray, num_bkps: int, jump: int, min_size: int
+    ):
+        """Initialize the DynamicProgramming instance with given parameters."""
+        self.dynp = _DynP.DynamicProgramming(data, num_bkps, jump, min_size)
+
+    def set_num_threads(self, num_threads: int):
```

Comment:
Long term dev wise, if additional CPP methods will be added we will most likely want num_threads to be controlled on the level of the whole module? Would preprocessor be a good place to have this? @b-butler  thoughts?

Reply:
Adding this to as a `_DynP` module function and exporting it to `dupin/util.py` is probably the best solution in my opinion.

Reply:
so I would just move this function to util.py right? would just have to import _DynP in util.py?

Reply:
Yes